ecadlabs/taqueria
# 🛠️ Fix ➾ Recompute ephemeral state after installing/uninstalling a plugin

# 🌮 Taqueria PR

## 🪁 Description

Before this PR, you couldn't do this:
```
taq init test-project && cd test-project && taq install @taqueria/plugin-ligo && taq compile --help
```

You'll see the generic help output rather than contextual help.

The reason why this fails is because there's a timing issue between installing a plugin and then invoking a task from that newly installed plugin.

*Please see the testing notes below*

## 🪂 Pre-Merge Checklist (Definition of Done)

🚦 Required to merge:

- [ ] ⛱️ I have completed this PR template in full and updated the title appropriately
- [ ] ⛵ My code builds cleanly, and I have manually tested the changes
- [ ] 🏄‍♂️ Another team member has built this branch and done manual testing on the change
- [ ] 🏖️ New and existing unit tests pass locally and in CI
- [ ] 🔱 The test plan has been implemented and verified by an SDET
- [ ] 🦀 Automated tests have been written and added to this PR
- [ ] 🤿 Corresponding changes have been made to all documentation
- [ ] 🐚 Required changes to the VScE have been made
- [ ] 🪸 Required updates to scaffolds have been made
- [ ] 🚢 The release checklist has been completed

## 🛩️ Summary of Changes

After installing/uninstalling a plugin, we remove the ephemeral state file and recompute it.

## 🎢 Test Plan

1. Using `taq` from main, verify that executing the following does _not_ display contextual help output for the ligo compile task: `taq init test-project && cd test-project && taq install @taqueria/plugin-ligo && taq compile --help`
2. Checkout mk-1666-e2e-for-flextesa-plugin
3. Merge in fix-plugin-installation-timing
4. Run `npm run build:binary`
5. Re-execute the following and verify that contextual help for the ligo compile task is displayed: `taq init test-project && cd test-project && taq install @taqueria/plugin-ligo && taq compile --help`
